be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -192,16 +192,19 @@ ProcessHashTableProbe<JoinOpType, ignore_null>::ProcessHashTableProbe(HashJoinNo
         : _join_node(join_node),
           _batch_size(batch_size),
           _build_blocks(join_node->_build_blocks),
-          _tuple_is_null_left_flags(
-                  reinterpret_cast<ColumnUInt8&>(*join_node->_tuple_is_null_left_flag_column)
-                          .get_data()),
-          _tuple_is_null_right_flags(
-                  reinterpret_cast<ColumnUInt8&>(*join_node->_tuple_is_null_right_flag_column)
-                          .get_data()),
           _rows_returned_counter(join_node->_rows_returned_counter),
           _search_hashtable_timer(join_node->_search_hashtable_timer),
           _build_side_output_timer(join_node->_build_side_output_timer),
-          _probe_side_output_timer(join_node->_probe_side_output_timer) {}
+          _probe_side_output_timer(join_node->_probe_side_output_timer) {
```

Comment:
Prefer to `_tuple_is_null_left_flags(join_node->_is_outer_join?reinterpret_cast<ColumnUInt8&>(*join_node->_tuple_is_null_left_flag_column)
                          .get_data():nullptr)`